### PR TITLE
Added basic scheduler implementation, unit tests and matchers.

### PR DESCRIPTION
**Pull request by frankscholten**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -34,8 +34,26 @@
...
     </properties>
 
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.2</version>
+                <configuration>
+                    <source>1.8</source>
```
Comment (mwl) [inline pom.xml]: Are we going for language level 8? Other Elastic plugins are still at 7.

File: scheduler/pom.xml [hunk 2]
```diff
@@ -24,6 +24,34 @@
...
+
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-all</artifactId>
+            <version>1.3</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>joda-time</groupId>
```
Comment (mwl) [inline scheduler/pom.xml]: If settling with JDK 8 there shouldn't be any need for joda-time. `java.time` should have everything.

File: pom.xml (hunk 1 above)
Comment (frankscholten) [inline pom.xml]: Let's do 7 then

**Comment by frankscholten** [comment]
Docker image for mesos/elasticsearch-cloud-mesos should be available soon.

**Comment by keithchambers** [comment]
It's would love to see a light weight image based on alpine. Something like https://github.com/soldair/docker-alpine-elasticsearch.

**Pull request merged**